lib/Common/Memory/Recycler.cpp

```diff
@@ -5674,8 +5731,57 @@ Recycler::FinishConcurrentCollect(CollectionFlags flags)
 #endif
         GCETW_INTERNAL(GC_STOP, (this, ETWEvent_ConcurrentRescan));
     }
+#if ENABLE_CONCURRENT_GC && ENABLE_ALLOCATIONS_DURING_CONCURRENT_SWEEP
+    else if (collectionState == CollectionStateConcurrentSweepPass1Wait)
+    {
+        this->FinishSweepPrep();
+        this->collectionState = CollectionStateConcurrentSweepPass2;
+
+        if (forceInThread)
+        {
+#ifdef RECYCLER_TRACE
+            if (this->GetRecyclerFlagsTable().Trace.IsEnabled(Js::ConcurrentSweepPhase))
+            {
+                Output::Print(_u("[GC #%d] Finishing Sweep Pass2 in-thread. \n"), this->collectionCount);
+            }
+#endif
+            this->recyclerSweep->FinishSweep();
+            this->FinishConcurrentSweep();
+            this->recyclerSweep->EndBackground();
+            //this->collectionState = CollectionStateConcurrentSweepPass2Wait;
+
+            uint sweptBytes = 0;
+#ifdef RECYCLER_STATS
+            sweptBytes = (uint)collectionStats.objectSweptBytes;
+#endif
+
+            GCETW(GC_BACKGROUNDSWEEP_STOP, (this, sweptBytes));
+            GCETW_INTERNAL(GC_STOP, (this, ETWEvent_ConcurrentSweep));
+
+            this->collectionState = CollectionStateTransferSweptWait;
+            RECYCLER_PROFILE_EXEC_BACKGROUND_END(this, Js::ConcurrentSweepPhase);
+
+            FinishTransferSwept(flags);
+        }
+        else
+        {
+            // Signal the background thread to finish concurrent sweep Pass2 for all the buckets.
+            SetEvent(this->concurrentWorkReadyEvent);
+            needConcurrentSweep = true;
```

Comment:
Probably should set this before SetEvent?

Reply:
Shouldn't matter. I have moved it up anyways.


---
In reply to: [148171378](https://github.com/Microsoft/ChakraCore/pull/4104#discussion_r148171378) [](ancestors = 148171378)